WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/ProductInOrderViewModel.swift

```diff
@@ -1,39 +1,71 @@
+    /// 
+    struct DiscountConfiguration {
+        let addedDiscount: Decimal
+        let baseAmountForDiscountPercentage: Decimal
+        let onSaveFormattedDiscount: (String?) -> Void
+    }
+
     /// The product being edited.
     ///
     let productRowViewModel: ProductRowViewModel
 
+    let addedDiscount: Decimal
+
+    var formattedDiscount: String? {
+        currencyFormatter.formatAmount(addedDiscount)
+    }
+
     let baseAmountForDiscountPercentage: Decimal
 
     /// Closure invoked when the product is removed.
     ///
     let onRemoveProduct: () -> Void
 
-    let isAddingDiscountToProductEnabled: Bool
+    private let isAddingDiscountToProductEnabled: Bool
+
+    var showAddDiscountRow: Bool {
+        isAddingDiscountToProductEnabled && addedDiscount == 0
+    }
+
+    var showCurrentDiscountSection: Bool {
+        isAddingDiscountToProductEnabled && addedDiscount != 0
+    }
 
     let onSaveFormattedDiscount: (String?) -> Void
 
+    var viewDismissPublisher = PassthroughSubject<(), Never>()
+
+    private let currencyFormatter: CurrencyFormatter
+
     init(productRowViewModel: ProductRowViewModel,
-         baseAmountForDiscountPercentage: Decimal,
-         onRemoveProduct: @escaping () -> Void,
-         onSaveFormattedDiscount: @escaping (String?) -> Void,
-         isAddingDiscountToProductEnabled: Bool = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.ordersWithCouponsM4)) {
+         productDiscountConfiguration: DiscountConfiguration?,
+         storeCurrencySettings: CurrencySettings = ServiceLocator.currencySettings,
+         onRemoveProduct: @escaping () -> Void) {
         self.productRowViewModel = productRowViewModel
-        self.baseAmountForDiscountPercentage = baseAmountForDiscountPercentage
+        self.addedDiscount = productDiscountConfiguration?.addedDiscount ?? .zero
+        self.baseAmountForDiscountPercentage = productDiscountConfiguration?.baseAmountForDiscountPercentage  ?? .zero
         self.onRemoveProduct = onRemoveProduct
-        self.onSaveFormattedDiscount = onSaveFormattedDiscount
-        self.isAddingDiscountToProductEnabled = isAddingDiscountToProductEnabled
+        self.onSaveFormattedDiscount = productDiscountConfiguration?.onSaveFormattedDiscount ?? { _ in }
+        self.isAddingDiscountToProductEnabled = productDiscountConfiguration != nil
+        self.currencyFormatter = CurrencyFormatter(currencySettings: storeCurrencySettings)
     }
 
     lazy var discountDetailsViewModel: FeeOrDiscountLineDetailsViewModel = {
-        FeeOrDiscountLineDetailsViewModel(isExistingLine: false,
+        FeeOrDiscountLineDetailsViewModel(isExistingLine: addedDiscount > 0,
                                           baseAmountForPercentage: baseAmountForDiscountPercentage,
-                                          initialTotal: "0.00",
+                                          initialTotal: formattedDiscount ?? "0",
```

Comment:
For the same reason as above (showing a $0.00 discount line when there's no discount), should we set this to empty string or nil rather than zero? Not sure if there are other implications I'm not taking into account 👍

Reply:
In this case `FeeOrDiscountLineDetailsViewModel` handles it naturally if we pass "0" as well. Semantically it also makes sense, when there is no discount the discount is 0.